src/plugins/intel_npu/src/plugin/src/compiled_model.cpp

```diff
@@ -128,17 +135,23 @@ std::shared_ptr<ov::ISyncInferRequest> CompiledModel::create_sync_infer_request(
 }
 
 void CompiledModel::export_model(std::ostream& stream) const {
-    const auto& blob = _networkPtr->compiledNetwork;
-    stream.write(reinterpret_cast<const char*>(blob.data()), blob.size());
+    std::stringstream str;
+    if (_networkPtr->compiledNetwork.size() == 0 && _networkPtr->metadata.graphHandle != nullptr) {
+        const auto blob = _compiler->getCompiledNetwork(_networkPtr);
```

Comment:
if this function returns compiledNetworks in both cases, not sure that we need an extra check here.

Reply:
yes, we have refactor this part in the last commit and remove the the condition check as well

  For CID path` getCompiledNetwork()` mean it get the blob from compiler through `pfnGetNativeBinary` using
  (networkDescription->metadata.graphHandle)

  For CIP path `getCompiledNetwork()` mean it get the blob form Icompiler's (networkDescription->compiledNetwork )